soc/arm/st_stm32/stm32f0/soc.h

```diff
@@ -55,6 +55,10 @@
 #include <stm32f0xx_ll_spi.h>
 #endif
 
+#ifdef CONFIG_GPIO_STM32
```

Comment:
Why not use CONFIG_USE_STM32_LL_GPIO instead?

It was created to flag the usage of LL_GPIO. Using CONFIG_GPIO_STM32 restricts the inclusion to the STM32_GPIO driver - but other drivers may also want to use the LL_GPIO functions.

Reply:
Understand, while I think it's because for other drivers, the interface of enabling gpio driver of stm32 should be always CONFIG_GPIO_STM32. Other drivers like spi drivers should use generic configuration functions to configure the port, pin, and the gpio driver will handle the configuration. No matter if the GPIO driver is using the LL library.
There is another reason for it, other drivers define CONFIG_SPI_STM32, CONFIG_I2C_STM32_V2... 
So, I'm new to zephyr and just expressing what I'm thinking about this problem, and I think this may should be discussed in a new thread.

Reply:
@b0661 , USE_STM32_LL_XXX is made for LL sources, not headers.
Using these flags in soc.h would work but it's a new way of using them and this deserve a dedicated discussion.